### ISSUE TITLE
Variation of Floyd-Warshall algorithm for Widest Path

### ISSUE THREAD

**Issue by blargg** [petgraph/petgraph#460]
### Summary

Would like to add a new variation of the floyd-warshall algorithm that instead of calculating the shortest path between all nodes, it calculates the widest path between all notes. The width of a path being the smallest capacity of any edge on the path.

### Motivation

This adds a new algorithm that I don't think petgraph has right now.

### Details

* What code needs to change? New traits? Which modules?
  * This would add a new function in petgraph::algo::floyd_warshall. This would possibly generalize the floyd_warshall to fit both cases.

* Is this a new graph algorithm? Include links to papers, Wikipedia, or other
  implementations of this algorithm that exist.
  * mentioned in the description of the floyd-warshall page: https://en.wikipedia.org/wiki/Floyd%E2%80%93Warshall_algorithm
  * Looks like there is a pseudocode implementation on the page for the schultz method, under implementation: https://en.wikipedia.org/wiki/Schulze_method

* Are you willing to implement this yourself? Mentor someone else and help them
  implement it?
  * Yes, implement. The code change looks minor if I can figure out a good way to reuse code between the functions.

**Comment by oxarbitrage** [comment]
I think this is a good idea. One option is to add an argument `variant` to the `floyd_warshall()` function:

```
enum FloydWarshallVariant {
    ShortestPath,
    WidestPath,
}
```

and do all the coding in one function (if that makes sense, i didn't checked in detail the changes needed)

**Comment by blargg** [comment]
Yep, I think that should work. I'll work on the code so you (or anyone) can review.
